### ISSUE TITLE
nowcast archiving for SABv20a fails on Frontera due to failed memory allocation

### ISSUE THREAD

**Issue by jasonfleming** [StormSurgeLive/asgs#605]
When running the SABv20a mesh on Frontera with SWAN coupling on 4031 compute cores, the nowcast archiving (which includes the recomposition of a fulldomain SWAN hotstart file from subdomain SWAN hotstart files) fails due to the inability to allocate memory:
```
[2021-Jul-28-T22:57:24-0500] INFO: nowcast: asgs_main.sh: Initiating nowcast archival process, if any.
Operating system error: Cannot allocate memory
Memory allocation failed

Error termination. Backtrace:

Program received signal SIGSEGV: Segmentation fault - invalid memory reference.

Backtrace for this error:
Operating system error: Cannot allocate memory
Memory allocation failed

Error termination. Backtrace:
#0  0x2b9ee43648e0 in ???
#1  0x2b9ee43653c5 in ???
#2  0x2b9ee43655c2 in ???
#3  0x2b9ee4363e0a in ???
#4  0x2b9ee4550753 in ???
#5  0x2b9ee455fe37 in ???
#6  0x4083d1 in ???
#7  0x40dbfe in ???
#8  0x40dcfb in ???
#9  0x2b9ee48d8554 in ???
#10  0x402208 in ???
#11  0xffffffffffffffff in ???
```
The short term workaround is to disable the recomposition of SWAN hotstart files in the archive script, which happens to be `enstorm_pedir_removal.sh`. The long term solution is for this process to occur in a batch job rather than a head node.

**Comment by wwlwpd** [comment]
segfault - what version of ADCIRC? is it just this grid?

**Comment by jasonfleming** [comment]
Hey @wwlwpd sorry for the delay, it is actually the `unhcat.exe` in SWAN that decomposes/recomposes the SWAN hotstart files. I think it is a memory hog, and this mesh is so big that it can (or isn't allowed) to allocate enough memory on the head node of Frontera to actually function. I think the solution to this is to have it run as a batch process.

**Comment by wwlwpd** [comment]
I see, thank you. Sounds like this process needs to be turned into a large memory serial job.

**Comment by wwlwpd** [comment]
_Note:_

11066018 elements
5584241 nodes

**Comment by wwlwpd** [comment]
@jasonfleming - so is the ask here to create support for a serial job if `WAVES="on"`? I'd to give that a shot if this is what we think we'd want, given _high memory_ requirement.

**Comment by jasonfleming** [comment]
There is a lot to keep track of with SWAN, I think we need a SWAN tag for issues, like we currently do for OPeNDAP. :-) 

However in this instance, I see a bigger strategic issue or architectural challenge. Basically, I would like to have a more generic mechanism for easily taking any subtask that the ASGS would normally execute on the head node and having it execute as a batch job. The manipulation of SWAN hotstart files described above would already be a batch job if we had a ready made mechanism executing it that way.

Related to this, it would also be nice if all the "prep" jobs could be folded into a single queue script as a serial job. We currently execute `adcprep` with `--partmesh` then `--prepall` or `--prep15` and sometimes `--prep20` each as their own batch jobs with inevitable delays in between. It would be nice if the ASGS could figure out what it needed and then run all these executables in a single queue script. 

Same goes for post-job tasks, like this SWAN hotstart file recomposition. Would be great to have this running alongside other clean up, post processing, and archiving tasks in a single batch job.

In summary, I see this task as a symptom of a larger scale issue that will require some refactoring to solve.

**Comment by wwlwpd** [comment]
ah good - thanks, this gives me something to think more about

**Comment by jasonfleming** [comment]
This will be addressed in the refactoring of our adcprep infrastructure.

**Comment by jasonfleming** [comment]
At a minimum, we need to have more fine grained control over SWAN hotstart recomposition in the ASGS config file.
